src/main/java/org/openstreetmap/atlas/checks/validation/relations/OneMemberRelationCheck.java

```diff
@@ -55,20 +59,54 @@ public boolean validCheckForObject(final AtlasObject object)
         // If the number of members in the relation is 1
         if (members.size() == 1)
         {
+            if (members.get(0).getEntity().getType().equals(ItemType.RELATION))
+            {
+                return Optional.of(createFlag(getRelationMembers((Relation) object),
+                        this.getLocalizedInstruction(2, relation.getOsmIdentifier(),
+                                members.get(0).getEntity().getOsmIdentifier())));
+            }
             // If the relation is a multi-polygon,
             if (relation.isMultiPolygon())
             {
-                return Optional.of(createFlag(
-                        relation.members().stream().map(RelationMember::getEntity)
-                                .collect(Collectors.toSet()),
+                return Optional.of(createFlag(getRelationMembers((Relation) object),
                         this.getLocalizedInstruction(1, relation.getOsmIdentifier())));
             }
-
-            return Optional.of(createFlag(
-                    relation.members().stream().map(RelationMember::getEntity)
-                            .collect(Collectors.toSet()),
+            return Optional.of(createFlag(getRelationMembers((Relation) object),
                     this.getLocalizedInstruction(0, relation.getOsmIdentifier())));
         }
         return Optional.empty();
     }
+
+    /**
+     * Gathers all the members of a relation, replacing child relations with their members.
+     *
+     * @param relation
+     *            {@link Relation} to get the members of
+     * @return A {@link Set} of the relations members as {@link AtlasObject}s
+     */
+    private Set<AtlasObject> getRelationMembers(final Relation relation)
```

Comment:
This looks like pretty valuable functionality - essentially it's flattening the relation tree structure. Can you add an atlas issue requesting this functionality? This is definitely a re-usable API.